src/backing/client/public_key.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef KMSENGINE_BACKING_CLIENT_PUBLIC_KEY_H_
+#define KMSENGINE_BACKING_CLIENT_PUBLIC_KEY_H_
+
+#include <string>
+
+#include "src/backing/client/crypto_key_version_algorithm.h"
+
+namespace kmsengine {
+namespace backing {
+
+// The public key for a Cloud KMS key.
+//
+// Used in lieu of `google::cloud::kms::v1::PublicKey` from the Cloud KMS API
+// protobuf definitions since the bridge layer needs to refer to this resource
+// directly and the bridge layer is not able to include external dependencies
+// (such as the generated protobuf definitions).
+class PublicKey {
+ public:
+  explicit PublicKey(std::string pem, CryptoKeyVersionAlgorithm algorithm)
+      : pem_(std::move(pem)), algorithm_(algorithm) {}
```

Comment:
Are you sure you want to `move` this?

Copying is safer, and the performance difference is probably not worth this micro-optimization.

Reply:
Actually, if the function argument type is `std::string`, the copy is done when arguments are passed into the `PublicKey` constructor. I don't think this extra `std::move` in the `pem_` constructor does anything. It may actually interfere with the compiler's own optimizations.